SU2_CFD/include/variables/CDiscAdjFEABoundVariable.hpp

```diff
@@ -46,69 +47,93 @@
  * \author R. Sanchez.
  * \version 6.2.0 "Falcon"
  */
-class CDiscAdjFEABoundVariable : public CDiscAdjFEAVariable {
-protected:
+class CDiscAdjFEABoundVariable final : public CDiscAdjFEAVariable {
+private:
 
-  su2double *FlowTraction_Sens;         /*!< \brief Adjoint of the flow tractions. */
-  su2double *SourceTerm_DispAdjoint;    /*!< \brief Source term applied into the displacement adjoint
-                                                    coming from external solvers. */
+  MatrixType FlowTraction_Sens;        /*!< \brief Adjoint of the flow tractions. */
+  MatrixType SourceTerm_DispAdjoint;   /*!< \brief Source term applied into the displacement
+                                                   adjoint coming from external solvers. */
 
-public:
-
-  /*!
-   * \brief Constructor of the class.
-   */
-  CDiscAdjFEABoundVariable(void);
+  CVertexMap<unsigned> VertexMap;  /*!< \brief Object that controls accesses to the variables of this class. */
 
+public:
   /*!
    * \overload
-   * \param[in] val_fea - Values of the fea solution (initialization value).
-   * \param[in] val_nDim - Number of dimensions of the problem.
-   * \param[in] val_nvar - Number of variables of the problem.
+   * \param[in] disp - Pointer to the adjoint value (initialization value).
+   * \param[in] vel - Pointer to the adjoint value (initialization value).
+   * \param[in] accel - Pointer to the adjoint value (initialization value).
+   * \param[in] npoint - Number of points/nodes/vertices in the domain.
+   * \param[in] ndim - Number of dimensions of the problem.
+   * \param[in] nvar - Number of variables of the problem.
+   * \param[in] unsteady - Allocate velocity and acceleration.
    * \param[in] config - Definition of the particular problem.
    */
-  CDiscAdjFEABoundVariable(su2double *val_fea, unsigned short val_nDim, unsigned short val_nvar, CConfig *config);
+  CDiscAdjFEABoundVariable(const su2double *disp, const su2double *vel, const su2double *accel,
+                           unsigned long npoint, unsigned long ndim, unsigned long nvar, bool unsteady, CConfig *config);
 
   /*!
    * \brief Destructor of the class.
    */
-  ~CDiscAdjFEABoundVariable(void);
+  ~CDiscAdjFEABoundVariable() = default;
+  
+  /*!
+   * \brief Allocate member variables for points marked as vertex (via "Set_isVertex").
+   * \param[in] config - Definition of the particular problem.
+   */
+  void AllocateBoundaryVariables(CConfig *config);
 
   /*!
    * \brief Set the FSI force sensitivity at the node
    * \param[in] iDim - spacial component
    * \param[in] val - value of the Sensitivity
    */
-  inline void SetFlowTractionSensitivity(unsigned short iDim, su2double val) { FlowTraction_Sens[iDim] = val; }
+  inline void SetFlowTractionSensitivity(unsigned long iPoint, unsigned long iDim, su2double val) override {
```

Comment:
Is there any particular reason why all of the arguments that were previously `unsigned short` are now `unsigned long` ?

Reply:
unsigned long is the index type of the container for both row and column indexes, since those indexes are used to compute the linear index on each access, I did not think it was prudent to have different types. I thought it was better to make that consistent across the interface.